include/l2/upper_mac.hpp

```diff
@@ -32,122 +24,28 @@ class UpperMac {
-    void fragmentation_start_burst();
-    void fragmentation_end_burst();
-    void fragmentation_push_tm_sdu_start(AddressType address_type, BitVector& vec);
-    void fragmentation_push_tm_sdu_frag(BitVector& vec);
-    void fragmentation_push_tm_sdu_end(BitVector& vec);
-    void fragmentation_push_tm_sdu_end_hu(BitVector& vec);
-
-    void remove_fill_bits(BitVector& vec);
-    bool remove_fill_bits_{};
-
-    // SYSINFO PDU
-    bool system_info_received_ = false;
-    int32_t downlink_frequency_ = 0;
-    int32_t uplink_frequency_ = 0;
-    uint8_t number_secondary_control_channels_main_carrier_{};
-    uint8_t ms_txpwr_max_cell_{};
-    uint8_t rxlev_access_min_{};
-    uint8_t access_parameter_{};
-    uint8_t radio_downlink_timeout_{};
-    uint8_t hyper_frame_cipher_key_flag_{};
-    uint8_t hyper_frame_number_{};
-    uint8_t common_cipher_key_identifier_or_static_cipher_key_version_number_{};
-    std::optional<uint32_t> even_multi_frame_definition_for_ts_mode_;
-    std::optional<uint32_t> odd_multi_frame_definition_for_ts_mode_;
-
-    struct DefaultDefinitionForAccessCodeA {
-        bool system_info_ = false;
-        uint8_t immediate_ = 0;
-        uint8_t waiting_time_ = 0;
-        uint8_t number_of_random_access_transmissions_on_up_link_ = 0;
-        uint8_t frame_length_factor_ = 0;
-        uint8_t timeslot_pointer_ = 0;
-        uint8_t minimum_pdu_priority_ = 0;
-    } defaults_for_access_code_a_;
-
-    struct ExtendedServiceBroadcast {
-        // TODO: I dont know how this works but maybe smart making for the sections
-        //  separate structs which then live in std::optionals
-        bool system_info_ = false;
-        uint8_t security_information_ = 0;
-        uint8_t sdstl_addressing_method_ = 0;
-        uint8_t gck_supported_ = 0;
-
-        bool system_info_section_1_ = false;
-        uint8_t data_priority_supported_ = 0;
-        uint8_t extended_advanced_links_and_max_ublck_supported_ = 0;
-        uint8_t qos_negotiation_supported_ = 0;
-        uint8_t d8psk_service_ = 0;
-
-        bool system_info_section_2_ = false;
-        uint8_t service_25Qam_ = 0;
-        uint8_t service_50Qam_ = 0;
-        uint8_t service_100Qam_ = 0;
-        uint8_t service_150Qam_ = 0;
-    } extended_service_broadcast_;
-
-    // STCH
-    bool second_slot_stolen_{};
-
-    std::shared_ptr<MobileLinkEntity> mobile_link_entity_{};
-    std::unique_ptr<LogicalLinkControl> logical_link_control_{};
-
-    // hashmap to keep track of framented mac segments
-    std::unordered_map<AddressType, std::vector<BitVector>> fragment_map_ = {};
-    std::vector<BitVector> fragment_list_{};
-    bool fragment_end_received_{};
-    bool fragment_end_hu_received_{};
-    AddressType last_address_type_{};
-    // save the last MAC-ACCESS or MAC-DATA where reservation_requirement is 0b0000 (1 sublot) for END-HU
-    AddressType last_address_type_end_hu_{};
-};
+    std::shared_ptr<Reporter> reporter_;
```

Comment:
why shared?

Reply:
this is due to older code architecture. It needs to be refactored in another PR